From 67c3fb57cf0b991c4f70aec78bdf25a47fb590b4 Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Fri, 2 Nov 2018 17:24:42 +0000
Subject: [PATCH 04/23] livepatch, fix: use THIS modname as the name of
 ddebug_table

We just want a unique name for every module, so put a _ddebug in
this file as the first of _ddebug array. Then remove path will work
correctly, use the mod->name.

Suggested-by: Li Bin <huawei.libin@huawei.com>
Signed-off-by: Zhou Chengming <zhouchengming1@huawei.com>
Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kmod/patch/livepatch-patch-hook.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kmod/patch/livepatch-patch-hook.c b/kmod/patch/livepatch-patch-hook.c
index e12fd50..6ae40de 100644
--- a/kmod/patch/livepatch-patch-hook.c
+++ b/kmod/patch/livepatch-patch-hook.c
@@ -473,6 +473,7 @@ out:
 
 static void __exit patch_exit(void)
 {
+	pr_debug("make THIS modname first\n");
 #ifndef HAVE_SIMPLE_ENABLE
 	WARN_ON(klp_unregister_patch(lpatch));
 #endif
-- 
2.18.1

